pkg/alertmanager/multitenant.go

```diff
@@ -806,12 +808,14 @@ type amConfig struct {
 // setConfig applies the given configuration to the alertmanager for `userID`,
 // creating an alertmanager if it doesn't already exist.
 func (am *MultitenantAlertmanager) setConfig(cfg amConfig) error {
-	// Instead of using "config" as the origin, as in Prometheus Alertmanager, we use "tenant".
-	// The reason for this that the config.Load function uses the origin "config",
-	// which is correct, but Mimir uses config.Load to validate both API requests and tenant
-	// configurations. This means metrics from API requests are confused with metrics from
-	// tenant configurations. To avoid this confusion, we use a different origin.
-	validateMatchersInConfigDesc(am.logger, "tenant", cfg.AlertConfigDesc)
+	if am.cfg.UTF8Migration {
```

Comment:
The reason I made this a separate feature flag instead of `!am.cfg.UTF8StrictMode` is because I believe some users might want to keep running this logging for a little while after enabling UTF-8 strict mode just to be 100% sure all their tenant configurations are OK. If I use `!am.cfg.UTF8StrictMode` then this use case is no longer possible.